minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/MiNiFiCommandSender.java

```diff
@@ -0,0 +1,131 @@
+    }
+
+    public Optional<String> sendCommand(String cmd, Integer port, String... extraParams) throws IOException {
+        Optional<String> response = Optional.empty();
+
+        if (port == null) {
+            LOGGER.info("Apache MiNiFi is not currently running");
+            return response;
+        }
+
+        try (Socket socket = new Socket()) {
+            LOGGER.debug("Connecting to MiNiFi instance");
+            socket.setSoTimeout(SOCKET_TIMEOUT);
+            socket.connect(new InetSocketAddress("localhost", port), CONNECTION_TIMEOUT);
+            LOGGER.debug("Established connection to MiNiFi instance.");
+
+            LOGGER.debug("Sending {} Command to port {}", cmd, port);
+
+            String responseString;
+            try (OutputStream out = socket.getOutputStream()) {
+                out.write(getCommand(cmd, extraParams));
+                out.flush();
+                responseString = readResponse(socket);
+            }
+
+            LOGGER.debug("Received response to {} command: {}", cmd, responseString);
+            response = Optional.of(responseString);
+        } catch (EOFException | SocketTimeoutException e) {
+            String message = "Failed to get response for " + cmd + " Potentially due to the process currently being down (restarting or otherwise)";
+            throw new RuntimeException(message);
+        }
+        return response;
+    }
+
+    <T> T sendCommandForObject(String cmd, Integer port, Class<T> clazz, String... extraParams) throws IOException {
+        return sendCommand(cmd, port, extraParams)
+            .map(response -> deserialize(cmd, response, clazz))
+            .orElse(null);
+    }
+
+    private String readResponse(Socket socket) throws IOException {
+        StringBuilder sb = new StringBuilder();
+        int numLines = 0;
+        try (BufferedReader reader = new BufferedReader(new InputStreamReader(socket.getInputStream()))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                if (numLines++ > 0) {
+                    sb.append("\n");
+                }
+                sb.append(line);
+            }
+        }
+
+        return sb.toString().trim();
+    }
+
+    private byte[] getCommand(String cmd, String... args) {
+        String argsString = Arrays.stream(args).collect(Collectors.joining(" "));
+        String commandWithArgs = cmd + " " + miNiFiParameters.getSecretKey() + (args.length > 0 ? " " : "") + argsString + "\n";
+        return commandWithArgs.getBytes(StandardCharsets.UTF_8);
+    }
+
+    private <T> T deserialize(String cmd, String obj, Class<T> clazz) {
+        T response;
+        try {
+            response = objectMapper.readValue(obj, clazz);
+        } catch (JsonProcessingException e) {
+            String message = "Failed to deserialize " + cmd + " response";
+            LOGGER.error(message);
+            throw new RuntimeException(message);
```

Comment:
The `JsonProcessingException` should be passed as the cause.
```suggestion
            throw new RuntimeException(message, e);
```

Reply:
Added